plugin/trino-spooling-filesystem/src/main/java/io/trino/spooling/filesystem/FileSystemSpoolingManagerFactory.java

```diff
@@ -40,20 +41,22 @@ public SpoolingManager create(Map<String, String> config, SpoolingManagerContext
     {
         requireNonNull(config, "requiredConfig is null");
 
-        Bootstrap app = new Bootstrap(
-                new FilesystemSpoolingModule(),
-                new MBeanModule(),
-                new MBeanServerModule(),
-                binder -> {
-                    binder.bind(SpoolingManagerContext.class).toInstance(context);
-                    binder.bind(OpenTelemetry.class).toInstance(context.getOpenTelemetry());
-                });
-
-        Injector injector = app
-                .doNotInitializeLogging()
-                .setRequiredConfigurationProperties(config)
-                .initialize();
-
-        return injector.getInstance(SpoolingManager.class);
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(FileSystemSpoolingManager.class.getClassLoader())) {
```

Comment:
This is not needed as `SpoolingManagerRegistry` already does this. For plugin factory `create()` methods, we do implement the wrappering in the engine (unlike for the instances).

Reply:
Without it this code breaks

Reply:
Can you share how it breaks?

Reply:
ClassNotFound exception

Reply:
which class?

Reply:
Mhm, it works now. Dropped.